src/stub/parser.rs

```diff
@@ -1,32 +1,29 @@
 #![allow(clippy::while_let_on_iterator)]
 
 pub mod types;
+
 pub use types::{Cmd, JoinTerm, JoinTermType, Stub, VariableCommand};
-mod token_stream;
 
 use regex::Regex;
-use token_stream::TokenStream;
+use std::iter;
 
 pub fn parse_generator_stub(generator: &str) -> Stub {
-    Parser::from(generator).parse()
+    Parser::new(generator).parse()
 }
 
 /// A wrapper around an iterator of tokens in the CG stub. Contains all of the stub parsing logic.
 ///
 /// Exists solely to be consumed with `.parse()`
 pub struct Parser<'a> {
-    token_stream: TokenStream<'a>,
+    token_stream: Box<dyn Iterator<Item = &'a str> + 'a>,
```

Comment:
Has to be a box since for convenience I want to simply pass this function an `&str` and have it do its thing.

The previous version was creating the iterator at the caller so the `StreamType` could be known by the compiler. But since the iterator is now created within `new` (where I feel it logically belongs) I would have to annotate the callers.